syntax = "proto3";

package authservice;

message ServerError {
    enum ErrorType {
        NO_ERROR = 0;
        USERNAME_ALREADY_EXISTS = 1;
        USERNAME_DOES_NOT_EXIST = 2;
        INVALID_USERNAME = 3;
        INVALID_PASSWORD = 4;
        INVALID_EMAIL = 5;
        INCORRECT_PASSWORD = 6;
        UNEXPECTED_ERROR = 7;
    }
  
    ErrorType type = 2;
    string message = 3;
};

service AuthService {
  rpc Login(LoginRequest) returns (LoginResponse) {}
  rpc Signin(SigninRequest) returns (SigninResponse) {}
}

message LoginRequest {
    string username = 1;
    string password = 2;
};

message LoginResponse {
    ServerError error = 1;
};

message SigninRequest {
    string username = 1;
    string password = 2;
    string confirmationPassword = 3;
    string email = 4;
};

message SigninResponse {
    ServerError error = 1;
};